string model_name
---
string model_type
string model_reference